### PR TITLE
Move DeletePossibleKicLeftOver function to delete pkg

### LINKED ISSUE #8040: Move logic out of `cmd/delete.go` so that it can be called from other packages
`cmd/delete.go` has a lot of interesting logic for cleaning up that is not accessible to `machine/delete.go`, and thus can't be used for cleanup on failure. 

Specifically, it allows for cleanup of orphaned KIC drivers - but we should also handle cleanup for orphaned non-KIC drivers.

### PR DESCRIPTION
**Pull request by BLasan**
fixes #8040 

Moved DeletePossibleKicLeftOver function from cmd/minikube/delete.go to a new package "delete" in order to make the function accessible from other packages.

### CODE REVIEW

**Comment by BLasan** [comment]
@medyagh @tstromberg  Please review

**Review by medyagh — CHANGES_REQUESTED** [review]
I dont agree with the naming, the package naming should not stutter.

delete.DeletePossibe... doesn't sounds right

**Comment by BLasan** [comment]
> I dont agree with the naming, the package naming should not stutter.
> 
> delete.DeletePossibe... doesn't sounds right

Yuh, is it ok to change the function name as well? I'll change the function name in to a meaningful one :)

**Comment by BLasan** [comment]
> I dont agree with the naming, the package naming should not stutter.
> 
> delete.DeletePossibe... doesn't sounds right

Renamed. Please review

**Comment by BLasan** [comment]
@medyagh Please review

**Comment by BLasan** [comment]
@medyagh Sir, could you please verify this? If there are changes need to be done I would like to add them as well :)

**Comment by BLasan** [comment]
@afbjorklund Sir, could you please verify these changes?

**Review by medyagh — CHANGES_REQUESTED** [review]
@BLasan what do u think about making a new package that is a noun

for example "profile"

so the could would read as

profile.DeletePossibleLeftovers()

**Comment by BLasan** [comment]
> @BLasan what do u think about making a new package that is a noun
> 
> for example "profile"
> 
> so the could would read as
> 
> profile.DeletePossibleLeftovers()

That sounds ideal sir! Before that I think we need to merge new changes done in the issue #10332. According to your suggestion above, isn't it good if we could define all the profile related functions(which are defined in "cmd/delete.go" file) under "profile" package?

**Comment by BLasan** [comment]
@medyagh @priyawadhwa Resolved the merge conflicts. Kindly take a look

File: pkg/minikube/profile/profile.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"os/exec"
+
+	"k8s.io/klog"
+	"k8s.io/minikube/pkg/drivers/kic/oci"
+	"k8s.io/minikube/pkg/minikube/driver"
+	"k8s.io/minikube/pkg/minikube/out"
+	"k8s.io/minikube/pkg/minikube/style"
+)
+
+//DeletePossibleLeftOvers ...
```
Comment (priyawadhwa) [inline pkg/minikube/profile/profile.go:31]: can you fill in this comment?

File: pkg/minikube/profile/profile.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*
```
Comment (priyawadhwa) [inline pkg/minikube/profile/profile.go:1]: Let's name this file `delete.go` instead of `profile.go`

**Review by priyawadhwa — COMMENTED** [review]
Left a couple comments, this is pretty close to finished!

File: pkg/minikube/profile/profile.go (hunk 1 above)
Comment (BLasan) [reply pkg/minikube/profile/profile.go:31]: Sure

**Comment by BLasan** [comment]
@priyawadhwa Changes added. Please review ma'am

**Comment by BLasan** [comment]
@priyawadhwa Any remaining things to be added ma'am?

File: go.mod [hunk 3]
```diff
@@ -96,6 +96,7 @@ require (
 	k8s.io/api v0.18.8
 	k8s.io/apimachinery v0.18.8
 	k8s.io/client-go v0.18.8
+	k8s.io/klog v1.0.0
```
Comment (priyawadhwa) [inline go.mod:99]: We want to use k8s.io/klog/v2 through the codebase, so to fix this I would:

1. Replace any imports of `k8s.io/klog` with `k8s.io/klog/v2`
2. Run `go mod tidy` to remove this dependency from go.mod

File: pkg/minikube/delete/delete.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+*/
+
+package delete
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+
+	"k8s.io/klog"
```
Comment (priyawadhwa) [inline pkg/minikube/delete/delete.go:24]: this should be k8s.io/klog/v2

**Review by priyawadhwa — COMMENTED** [review]
One last comment

**Comment by BLasan** [comment]
@priyawadhwa Thanks for the review ma'am. Made the required changes

**Review by priyawadhwa — APPROVED** [review]
LGTM, thank you!

**Comment by BLasan** [comment]
@medyagh @priyawadhwa Any remaining works?

**Pull request merged**